presto-main/src/main/java/com/facebook/presto/server/QuerySessionSupplier.java

```diff
@@ -124,6 +133,22 @@ public Session createSession(QueryId queryId, SessionContext context)
             sessionBuilder.setLocale(Locale.forLanguageTag(context.getLanguage()));
         }
 
+        if (sessionPropertyConfigurationManager.get() != null) {
+            SessionConfigurationContext configContext = new SessionConfigurationContext(
+                    context.getIdentity().getUser(),
+                    Optional.ofNullable(context.getSource()),
+                    context.getClientTags());
+            for (Entry<String, String> entry : sessionPropertyConfigurationManager.get().getSystemSessionProperties(configContext).entrySet()) {
```

Comment:
This can be simplified using `Map.forEach()`
```java
SessionPropertyConfigurationManager manager = sessionPropertyConfigurationManager.get();
manager.getSystemSessionProperties(configContext).forEach(sessionBuilder::setSystemProperty);
manager.getCatalogSessionProperties(configContext).forEach((catalog, properties) ->
    properties.forEach((key, value) -> sessionBuilder.setCatalogSessionProperty(catalog, key, value)));

Reply:
I hear different opinions on this in different occasions. Sometimes I'm advised to use for loop which is easier to read, other times I'm advised to use stream for shorter code. Is there a consistent guideline for this? Here I'd prefer keep the for loop because a few lines down, the old code that adds session properties from client session uses for loop. It's better to be consistent. And I don't see a strong enough argument to refactor the original.

Reply:
For a collection/iterable, I prefer using a for-each loop since it's easier to read. For maps, using `forEach()` can be substantially more readable since it gets rid of the verbose `Entry` type declaration and the need to unpack the types.

So you go from this
```java
for (Entry<String, Integer> entry : users) {
    String name = entry.getKey();
    String value = entry.getValue();
    ...
}
```

To this
```java
users.forEach((name, age) -> {
    ...
});
```

Much less boilerplate which makes it easier to focus on the logic.

Reply:
The code below would also benefit from using `Map.forEach()`

Reply:
But the consistency argument is reasonable, so we can leave this as-is and refactor the whole thing later